### DISCUSSION TITLE
Is there a event hook for listening abort/canceled event?

### DISCUSSION THREAD

**Discussion by joe223** [TanStack/query#7995] [Q&A] [answered]
Is there a event hook for listening abort/canceled event when the request get aborted 🤔?

for example, imagine there is a infinite scroll pagination component which provide a list to show items (it will triger `/api/list/:page_num`), and each item component will trying to load details if it in viewport (Yes, there will be another request to fetch `/api/details/:id`). we wanna item component can abort request if HTTP request not finished and this item scroll out of viewport (become invisible), also, to get the `onAbort` / `onSuccess` / `onError` event for performance metrics.

 <img src="https://github.com/user-attachments/assets/7b9b534c-120d-4837-997a-aeaf997312bc" alt="alt text" width="200"/>

I did not found the `onAbort` api, and `onError`/`onSuccess` will be **Deprecated**. So, is there any recommendations or best practices?

BTW, also curious why `onError`/`onSuccess` got deprecated 🙏?

**Comment by TkDodo** [comment, marked as answer]
only way I see is to subscribe to the queryCache (`queryClient.getQueryCache().subscribe`), then listen to the `update` event, then look at the `action.type`, which should be `error`, then check if the error is a cancelledError with `isCancelledError` (can be imported).
